src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -43,9 +43,33 @@ object EdgeProcessor extends StrictLogging {
       }
 
       cb.checkpoint.edge.data.messages.foreach { m =>
-        dao.messageService.put(m.signedMessageData.data.channelId,
-                               ChannelMessageMetadata(m, Some(cb.baseHash)))
-        dao.messageService.put(m.signedMessageData.signatures.hash,
+        if (m.signedMessageData.data.previousMessageHash != Genesis.CoinBaseHash) {
+          dao.messageService.put(
+            m.signedMessageData.data.channelName,
+            ChannelMessageMetadata(m, Some(cb.baseHash))
+          )
+          dao.channelService.updateOnly(
+            m.signedMessageData.hash,
+            { cmd =>
+              val slicedMessages = if (cmd.last25MessageHashes.size > 25) {
+                cmd.last25MessageHashes.slice(0, 24)
```

Comment:
You could always slice it 
`cmd.last25MessageHashes.slice(0, 24)` will work even collection size will be smaller

```
scala> val s = Seq(0,1,2,3)
s: Seq[Int] = List(0, 1, 2, 3)
scala> s.slice(0,25)
res2: Seq[Int] = List(0, 1, 2, 3)
```